src/sponge.rs

```diff
@@ -0,0 +1,7 @@
+use crate::{JubJubAffine, JubJubExtended, JubJubScalar};
+use poseidon252::sponge::sponge::sponge_hash;
+
+pub fn hash(p: &JubJubExtended) -> JubJubScalar {
+    let p = JubJubAffine::from(p);
+    JubJubScalar::from_raw(sponge_hash(&[p.get_x(), p.get_y()]).reduce().0)
```

Comment:
Why sponge mode if its easier to integrate into zk circuit a normal permutation and the width of 5 accepts these two inputs?

Reply:
Not sure I'm following, sorry. Could you rephrase?
Take in account that this doesn't need to be proved in zk, it's just an optimization to the previous hashing (and I double checked with Dmitry that the approach was sound).

Basically we want to hash an `AffinePoint` into a jubjub's scalar.  Since Poseidon takes BLS scalars, and the AffinePoint coords are BLS, it's easy doing so. The problem was the returned value that could be not a valid jubjub's scalar – but the function take care of it thanks to the `from_raw` + `reduce`.

Reply:
The approach is fine for now. But definitely we need to give it a thought on the future.